tests/unit/test_cli.py

```diff
@@ -12,100 +12,191 @@
-        (["--run", "--log-level", "warning"], True, "WARNING", True),
-        (["--log-level", "debug"], False, "DEBUG", True),
+        (0, "ERROR"),
+        (1, "WARNING"),
+        (2, "INFO"),
+        (3, "DEBUG"),
+        (4, "NOTSET"),
+        (5, "NOTSET"),
+        (10, "NOTSET"),
     ],
 )
-def test_parse_args(args, expected_run, expected_loglevel, expected_interactive):
-    parsed_args = parse_args(args)
-    assert parsed_args.run == expected_run
-    assert parsed_args.loglevel == expected_loglevel
-    assert parsed_args.interactive == expected_interactive
+def test_verbosity_level(verbosity_value, verbosity_name):
+    """Test VerbosityLevel Enum class."""
+    level = cli.VerbosityLevel(verbosity_value)
+    assert level.name == verbosity_name
 
 
 @pytest.mark.parametrize(
-    "args, exception", [(["--log-level=DDD"], ArgumentError), (["--foo"], SystemExit)]
+    "verbosity_value, exception", [(-1, ValueError), ("UNEXPECTED", ValueError)]
 )
-def test_parse_args_raise_exception(args, exception):
+def test_verbosity_level_exception(verbosity_value, exception):
+    """Test VerbosityLevel Enum class with invalid inputs."""
     with pytest.raises(exception):
-        parse_args(args)
+        cli.VerbosityLevel(verbosity_value)
 
 
-def test_setup_logging():
-    with patch("cou.cli.logging") as mock_logging:
-        log_file_handler = MagicMock()
-        console_handler = MagicMock()
-        mock_root_logger = mock_logging.getLogger.return_value
-        mock_logging.FileHandler.return_value = log_file_handler
-        mock_logging.StreamHandler.return_value = console_handler
-        setup_logging("INFO")
-        mock_root_logger.addHandler.assert_any_call(log_file_handler)
-        mock_root_logger.addHandler.assert_any_call(console_handler)
+@pytest.mark.parametrize(
+    "quiet, verbosity, level",
+    [(True, 0, "CRITICAL"), (True, 5, "CRITICAL"), (False, 0, "ERROR"), (False, 2, "INFO")],
+)
+def test_get_log_level(quiet, verbosity, level):
+    """Test get_log_level return value."""
+    assert cli.get_log_level(quiet=quiet, verbosity=verbosity) == level
 
 
 @pytest.mark.asyncio
 @pytest.mark.parametrize(
-    "exception, exp_exitcode",
+    "model_name, expected_call_count",
+    [("model_name", 0), ("", 1), (None, 1)],
+)
+async def test_analyze_and_plan(model_name, expected_call_count):
+    """Test analyze_and_plan function with different model_name arguments."""
+    analysis_result = Analysis(model_name=model_name, apps_control_plane=[], apps_data_plane=[])
+
+    with patch(
+        "cou.cli.juju_utils.get_current_model_name",
+        new=AsyncMock(return_value=model_name),
+    ) as mock_get_current_model_name, patch(
+        "cou.cli.generate_plan", new=AsyncMock()
+    ) as mock_generate_plan, patch(
+        "cou.cli.Analysis.create",
+        new=AsyncMock(return_value=analysis_result),
+    ) as mock_analyze:
+        await cli.analyze_and_plan(model_name)
```

Comment:
not-blocking: I think it's more readable to use `mocker` or `patch` as decorator.

Reply:
Replaced complex context manager with decorators